index.js

```diff
@@ -1,18 +1,55 @@
-import fetch from 'node-fetch';
-// Recommend using node-fetch for those familiar with JS fetch
-
-const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
-
-/**
- * @param name filter for color name
- * @param hex filter for color hex code
- * @param compName filter for complementary color name
- * @param compHex filter for complementary color hex code
- * @returns Promise
- */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+import fetch from "node-fetch";
+
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  try {
+    const response = await fetch("https://nt-cdn.s3.amazonaws.com/colors.json");
+    const colors = await response.json();
+
+    const filteredColors = colors.filter((color) => {
+      // check the name
+      const nameMatches = name
+        ? color.name.toLowerCase() === name.toLowerCase()
+        : false;
+
+      // check the hex #
+      const hexMatches = hex
+        ? color.hex.toUpperCase() === hex.toUpperCase()
+        : false;
+
+      // check the compName
+      const compNameMatches = compName
+        ? color.comp.some(
+            (compColor) =>
+              compColor.name.toLowerCase() === compName.toLowerCase()
+          )
+        : false;
+
+      // check the compHex #
+      const compHexMatches = compHex
+        ? color.comp.some(
+            (compColor) => compColor.hex.toUpperCase() === compHex.toUpperCase()
+          )
+        : false;
+
+      return hexMatches || nameMatches || compNameMatches || compHexMatches;
```

Comment:
if nameMatches, there's no need to do the other comparisons. in this case its not an issue, but its good to always think about performance.